types/vote.go

```diff
@@ -257,11 +279,9 @@ func (vote *Vote) ValidateBasic() error {
 
 func (ext VoteExtension) Copy() VoteExtension {
 	res := VoteExtension{
-		AppDataToSign:             make([]byte, len(ext.AppDataToSign)),
-		AppDataSelfAuthenticating: make([]byte, len(ext.AppDataSelfAuthenticating)),
+		AppDataToSign:             ext.AppDataToSign,
+		AppDataSelfAuthenticating: ext.AppDataSelfAuthenticating,
 	}
-	copy(res.AppDataToSign, ext.AppDataToSign)
-	copy(res.AppDataSelfAuthenticating, ext.AppDataSelfAuthenticating)
```

Comment:
Wasn't this right before, and is now not a new copy?

Reply:
It is weird that `copy` was not copying the byte slice into the fields and made error on the tests, but not using copy worked. There is no modification on the underlying byte slice AFAIK so I think this is okay.